Stop highlight after __DATA__ and __END__

Fank opened the pull request:
The two control characters `^D` and `^Z`, and the tokens `__END__` and `__DATA__` may be used to indicate the logical end of the script before the actual end of file. Any following text is ignored.

http://perldoc.perl.org/perldata.html#Special-Literals

kevinsawicki commented on grammars/perl.cson:
```diff
@@ -721,10 +721,17 @@
     'name': 'constant.language.perl'
   }
   {
+    'begin': '(__DATA__)\\n?'
```
Should this start with a `\\b` so that something like `FOO__DATA__` isn't mis-interpreted?

Fank replied:
Fixed thanks for the info

infininight commented:
Was looking at the documentation to see if this could be anchored to the line beginning but while that is generally the usage it doesn't appear that is required. Did see though that `__END__` should function the same in this context from a grammar perspective, could move that one as well.

Fank commented:
@infininight Thanks for the info i updated the Description with the info from the perl docs, and moved the `__END__` block to `__DATA__`.
What do you think about the control characters (in the description), should these be handled to?

kevinsawicki commented:
:ship: Looks great
